content/docs/15.how-to-guides/build-custom-plugin.md

```diff
@@ -0,0 +1,425 @@
+
+As you can see, there is one generated plugin: the `Example` class representing the `Example` plugin (a task).
+
+A project typically hosts multiple plugins. We call a project a group of plugins, and you can have multiple sub-groups inside a project by splitting plugins into different packages. Each package that has a plugin class is a sub-group of plugins.
+
+## Gradle Configuration
+
+We use [Gradle](https://gradle.org/) as a build tool.
+
+### Mandatory configuration
+
+The first thing we need to configure is the plugin name and the class package.
+
+1. Change in `settings.gradle` the `rootProject.name = 'plugin-template'` with the plugin name `rootProject.name = 'plugin-pokemon'`.
+2. Change the class package: by default, the template provides a package `io.kestra.plugin.templates`, just rename the `templates` folder in `src/main/java` & `src/test/java` to `pokemon`. And change the first line in `Example.java`, `ExampleRunnerTest.java` and `ExampleTest.java` to `package io.kestra.plugin.pokemon;`.
+3. In `build.gradle`: 
+   a. replace `description 'Plugin template for Kestra'` to the package name `description 'Plugin pokemon for Kestra'`.
+   b. In the `dependencies` section, add a dependency which we will be using in our plugin task:
+    `implementation group: 'com.googlecode.json-simple', name: 'json-simple', version: '1.1.1'` 
+   c. Change the `jar` section to the following:
+
+```groovy
+jar {
+    manifest {
+        attributes(
+                "X-Kestra-Name": project.name,
+                "X-Kestra-Title": "Pokemon",
+                "X-Kestra-Group": project.group + ".pokemon",
+                "X-Kestra-Description": project.description,
+                "X-Kestra-Version": project.version
+        )
+    }
+}
+```
+
+## Develop Fetch Task
+
+### Create Pokemon class
+
+In `src/main/java/io/kestra/plugin/pokemon`, we will create a new class `Pokemon.java`. This will be used to map the JSON output of the pokemon API to the Java class. We only need to add the fields that we are interested in, and ignore the rest.
+
+::collapse{title="Here is how the Pokemon.java file should look like"}
+```java
+package io.kestra.plugin.pokemon;
+
+import java.util.*;
+import lombok.*;
+
+@Data
+public class Pokemon {
+  List<DetailedAbility> abilities;
+  long base_experience;
+  long height;
+  List<DetailedMove> moves;
+}
+
+@Data
+class DetailedAbility {
+  Ability ability;
+}
+
+@Data
+class Ability {
+  String name;
+}
+
+@Data
+class DetailedMove {
+  Move move;
+}
+
+@Data
+class Move {
+  String name;
+}
+```
+::
+
+### Runnable Task
+
+We will refactor the java file from `Example.java` to `Fetch.java`.
+
+In this file, we will put in the appropriate schema for the plugin, and for the inputs and output of the plugin. This will help in generating proper documentation for the plugin. We will also put in examples so that any user can get help in understanding how to use the plugin.
```

Comment:
```suggestion
In this file, we will put in the appropriate schema for the plugin, and for the inputs and output of the plugin. This will help in generating proper documentation for the plugin. We will also put in examples so that any user can get help in understanding how to use the plugin.
```
```suggestion
In this file, we will put in the appropriate schema for the plugin, including the inputs and output of the plugin. This will help us generate documentation for the plugin too. Also, we will also include a few examples to help users understand how to use the plugin.
```